src/main/java/com/marklogic/mule/extension/Operations.java

```diff
@@ -112,66 +113,100 @@ public void writeDocuments(
-    public List<Result<InputStream, DocumentAttributes>> searchDocuments(
-        @Connection DatabaseClient databaseClient,
+    public PagingProvider<DatabaseClient, Result<InputStream, DocumentAttributes>> searchDocuments(
         @DisplayName("Collection") @Optional(defaultValue = "") @Example("myCollection") String collection,
         @DisplayName("Query") @Text @Optional @Example("searchTerm") String query,
         @DisplayName("Query Type") @Optional QueryType queryType,
         @DisplayName("Query Format") @Optional QueryFormat queryFormat,
         @DisplayName("Metadata Category List") @Optional(defaultValue = "all") @Example("COLLECTIONS,PERMISSIONS") String categories,
         @DisplayName("Max Results") @Optional @Example("10") Integer maxResults,
+        @DisplayName("Page Length") @Optional @Example("10") Integer pageLength,
         @DisplayName("Search Options") @Optional @Example("appSearchOptions") String searchOptions,
         @DisplayName("Directory") @Optional @Example("/customerData") String directory,
         @DisplayName("REST Transform") @Optional String restTransform,
         @DisplayName("REST Transform Parameters") @Optional String restTransformParameters,
         @DisplayName("REST Transform Parameters Delimiter") @Optional @Example(",") String restTransformParametersDelimiter
     ) {
-        DocumentManager<JSONReadHandle, JSONWriteHandle> documentManager = databaseClient.newJSONDocumentManager();
-        if (Utilities.hasText(categories)) {
-            documentManager.setMetadataCategories(buildMetadataCategories(categories));
-        }
-        if (maxResults != null) {
-            documentManager.setPageLength(maxResults);
+        final int finalPageLength;
+        if (pageLength != null) {
+            finalPageLength = pageLength;
+        } else {
+            finalPageLength = 10;
         }
 
-        QueryDefinition queryDefinition = ReadUtil.buildQueryDefinitionFromParams(databaseClient, query, queryType, queryFormat);
-        if (Utilities.hasText(collection)) {
-            queryDefinition.setCollections(collection);
-        }
-        if (Utilities.hasText(searchOptions)) {
-            queryDefinition.setOptionsName(searchOptions);
-        }
-        if (Utilities.hasText(directory)) {
-            queryDefinition.setDirectory(directory);
-        }
-        if (Utilities.hasText(restTransform)) {
-            ServerTransform serverTransform = new ServerTransform(restTransform);
-            if (Utilities.hasText(restTransformParameters)) {
-                String[] parametersArray = restTransformParameters.split(restTransformParametersDelimiter);
-                for (int i = 0; i < parametersArray.length; i = i + 2) {
-                    serverTransform.addParameter(parametersArray[i], parametersArray[i + 1]);
+        return new PagingProvider<DatabaseClient, Result<InputStream, DocumentAttributes>>() {
+            int totalRecordsDelivered = 0;
```

Comment:
Use "documents" instead of "records" since we know for certain we're returning documents ("records" is better when we need an abstraction over documents/rows/etc, but we know we're dealing with documents here). 

And `curPage` should be `currentPage` as we don't have any usage of "cur" elsewhere.